code/modules/atmospherics/FEA_gas_mixture.dm

```diff
@@ -151,10 +169,12 @@ What are the archived variables for?
 		. |= REACTION_ACTIVE
 
 	fuel_burnt = 0
-	if(temperature > FIRE_MINIMUM_TEMPERATURE_TO_EXIST)
-		if(fire() > 0)
+	if(src.temperature > FIRE_MINIMUM_TEMPERATURE_TO_EXIST)
+		if(fire())
```

Comment:
don't trust that this won't randomly become negative and ruin your day
```suggestion
		if(fire() > 0)
```

Reply:
better idea to runtime if it goes below 0 because that means we have burnt negative moles, which uh, bad

Reply:
ill put an assert

Reply:
You didn't put an assert